### PR TITLE
Fix slashes for paths containing non-ASCII characters on Windows.

### PR DESCRIPTION
**Pull request by Lifeni**
## Changes

Related issue: [#4689](https://github.com/withastro/astro/issues/4689)

The [`slash`](https://github.com/sindresorhus/slash) library [cannot handle paths containing non-ASCII characters](https://github.com/sindresorhus/slash/blob/b5cdd12272f94cfc37c01ac9c2b4e22973e258e5/index.js#L5), so we need to manually replace `\` with `/`.

#### Examples

```js
fileURLToPath(filePath)
// -> C:\Codes\测试\my-astro-site\src\pages\index.astro

slash(fileURLToPath(filePath))
// -> C:\Codes\测试\my-astro-site\src\pages\index.astro

slash(fileURLToPath(filePath)).replace(/\\/g, '/')
// -> C:/Codes/测试/my-astro-site/src/pages/index.astro ✅

```

## Testing

Test locally only.

## Docs

None.

### CODE REVIEW

**Comment by matthewp** [comment]
Thanks @Lifeni, can you add a test? If you look in `packages/astro/test/fixtures` you just create a mini project to test this. Would be wonderful

**Comment by Lifeni** [comment]
@matthewp Sorry, I'm new to pull requests and don't have much experience. It seems that I should not merge `withastro:main`, so what should I do?

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -3225,7 +3231,7 @@ packages:
     dependencies:
       '@astro-community/astro-embed-twitter': 0.1.3_astro@packages+astro
       '@astro-community/astro-embed-youtube': 0.1.1_astro@packages+astro
-      astro: link:packages/astro
+      astro: link:packages\astro
```
Comment (matthewp) [inline pnpm-lock.yaml:3234]: not sure why these changed, can you fix them?

File: packages/astro/test/non-ascii-path.test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		fixture = await loadFixture({ root: './fixtures/non-ascii-path/测试/' });
+		await fixture.build();
+	});
+
+	describe('build', () => {
+		it('Can load page', async () => {
+			const html = await fixture.readFile(`/index.html`);
+			const $ = cheerio.load(html);
+
+			expect($('h1').text()).to.equal('测试 OK');
```
Comment (matthewp) [inline packages/astro/test/non-ascii-path.test.js:18]: great looking test!

File: pnpm-lock.yaml (hunk 1 above)
Comment (Lifeni) [reply pnpm-lock.yaml:3234]: Thanks, I will fix them.

File: packages/astro/src/core/util.ts [hunk 3]
```diff
@@ -126,7 +126,7 @@ export function resolveDependency(dep: string, projectRoot: URL) {
  *   Windows:    C:/Users/astro/code/my-project/src/pages/index.astro
  */
 export function viteID(filePath: URL): string {
-	return slash(fileURLToPath(filePath) + filePath.search);
+	return slash(fileURLToPath(filePath) + filePath.search).replace(/\\/g, '/');
```
Comment (bluwy) [inline packages/astro/src/core/util.ts:129]: Do we still need to use `slash` if we're always doing replace ourselves now?

**Pull request merged**